src/content/docs/en/guides/deploy/github.mdx

```diff
@@ -130,40 +63,98 @@ export default defineConfig({
             uses: actions/deploy-pages@v4
     ```
 
-    :::note
-    The astro action takes a few optional inputs. These can be provided by uncommenting the `with:` line and the input you want to use.
-    :::
+     The astro action takes a few optional inputs. These can be provided by uncommenting the `with:` line and the input you want to use. Additionally, you will need to define any public environment variables. (See the [GitHub documentation on setting secrets](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#creating-configuration-variables-for-a-repository) for adding private environment variables.)
     
     :::caution
     The official Astro [action](https://github.com/withastro/action) scans for a lockfile to detect your preferred package manager (`npm`, `yarn`, `pnpm`, or `bun`). You should commit your package manager's automatically generated `package-lock.json`, `yarn.lock`, `pnpm-lock.yaml`, or `bun.lockb` file to your repository.
     :::
 
-2. (Optional) If you pass environment variables to your Astro project during local development or when previewing builds, you will need to define any public variables in the `deploy.yml` file in order for them to be processed when you deploy to GitHub Pages. (See the [GitHub documentation on setting secrets](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#creating-configuration-variables-for-a-repository) for adding private environment variables.)
+2. In your Astro config file `astro.config.mjs`, set [`site`](/en/reference/configuration-reference/#site) to the GitHub URL of your deployed site.
 
-   ```yaml title="deploy.yml"
-   jobs:
-      build:
-        runs-on: ubuntu-latest
-        steps:
-          - name: Checkout your repository using git
-            uses: actions/checkout@v4
-          - name: Install, build, and upload your site
-            uses: withastro/action@v3
-            env:
-              # Use single quotation marks for the variable value
-              PUBLIC_EVM_WALLET_ADDRESS: '0x4bFc229A40d41698154336aFF864f61083E76659'
-   ```
+    The value for `site` must be one of the following:
 
+    - The following URL based on your username: `https://<username>.github.io`
+    - The random URL autogenerated for a [GitHub Organization's private page](https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site): `https://<random-string>.pages.github.io/`
```

Comment:
Thoughts on having this _after_ the code block? Or do we think that’s confusing.

In some ways I’d like the direct opening line “Set `site`” followed by the code block setting `site`, with the bullet points following with more details. But then there’s a question of consistency with the next step, so I’ll leave that to you to decide.